multipart/multipart.py

```diff
@@ -1659,15 +1681,15 @@ def on_headers_finished() -> None:
 
                 # Create the proper class.
                 if file_name is None:
-                    f = FieldClass(field_name)
+                    f = FieldClass(field_name, headers=headers)
                 else:
-                    f = FileClass(file_name, field_name, config=self.config)
+                    f = FileClass(file_name, field_name, config=self.config, headers=headers)
```

Comment:
Should we instead just pass the `content_type` here? Is there any request to check other headers?

Reply:
Yes, good idea. This would also ease some "case-insensitive" complications.
I'll update the PR

Reply:
Thanks.

Also, thanks for the patience here. Sorry for the delay on review.

Reply:
Done.